core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java

```diff
@@ -288,6 +289,7 @@ private void runInternal() {
           CommonUtils.sleepMs(LOG, mJournalCheckpointSleepTimeMs);
         }
       }
+      // TODO(jiacheng): If we just return here due to interrupt incorrectly, does the master realize that?
       if (Thread.interrupted() && !mShutdownInitiated) {
```

Comment:
why would the thread interrupt without shut down? Master failover?

Reply:
https://github.com/Alluxio/alluxio/blob/master/core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java#L347-L349 
not sure but as FYI there's a place that the thread can interrupt itself.

Reply:
Seems https://github.com/Alluxio/alluxio/blob/master/core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java#L347-L349 is used to propagate the InterruptedException from deeper down the stack? I don't find a place where we wrap an InterruptedException in IOException yet at a glance. Seems we do use interrupt to stop the UfsJournalCheckpointThread https://github.com/Alluxio/alluxio/blob/b53129acaeb743728593403088b57425239df265/core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java#L147

The `mShutdownInitiated` is also a bit ambiguous, because it's used to mark the master shutting down as well as in `suspend()/resume()` used to pause the master when taking a backup. I guess the reasoning is: When the `mShutdownInitiated=true` then the thread is already shutting down, so we don't need to consider that in the `runInternal()`.

Reply:
> The `mShutdownInitiated` is also a bit ambiguous, because it's used to mark the master shutting down as well as in `suspend()/resume()` used to pause the master when taking a backup. I guess the reasoning is: When the `mShutdownInitiated=true` then the thread is already shutting down, so we don't need to consider that in the `runInternal()`.

In general, a lot of the designs in the journal were not created with the idea of gaining primacy / losing primacy or being paused and then resumed: most of the components were designed to be started once and run uninterrupted until they are stopped when the master is shut down. When the lose / gain primacy or pause / resume concepts were introduced later on, they piggy-backed off the already existing implementation, hence the many ambiguously named method and variables. These should be thoroughly reviewed for 3.x.